go/saltpack/decrypt.go

```diff
@@ -92,14 +92,30 @@ func (ds *decryptStream) read(b []byte) (n int, err error) {
 	return n, nil
 }
 
-func (ds *decryptStream) readHeader() error {
-	var hdr EncryptionHeader
-	seqno, err := ds.mps.Read(&hdr)
+func (ds *decryptStream) readHeader(rawReader io.Reader) error {
+	// Parse the length of the header.
+	var headerLength int
+	seqno, err := ds.mps.Read(&headerLength)
+	if err != nil {
+		return ErrFailedToDecodeHeaderLength
+	}
+	// Read the header bytes.
+	headerBytes := make([]byte, headerLength)
+	_, err = io.ReadFull(rawReader, headerBytes)
```

Comment:
I'm relying on the MessagePack stream decoder to only read exactly the bytes it needs, when it's decoding the frame int. If it over-reads, then we're going to get garbage. I don't think it's a security issue, but it could screw up an otherwise-valid message?

The alternative would be to do all of the initial reads by hand ourselves. We could do that with a little function that knew enough about MessagePack to read that int out of the stream, with exactly the reading behavior we want. Not sure it's worth it though.